torch/_inductor/lowering.py

```diff
@@ -3306,6 +3310,57 @@ def load_source_val():
     return view(result_flat, self.get_size())
 
 
+fallback__unsafe_masked_index = fallback_handler(
+    aten._unsafe_masked_index.default, add_to_fallback_set=False
+)
+
+fallback__unsafe_masked_index_put_accumulate = fallback_handler(
+    aten._unsafe_masked_index_put_accumulate.default, add_to_fallback_set=False
+)
+
+
+@register_lowering(aten._unsafe_masked_index, type_promotion_kind=None)
+def _unsafe_masked_index(self, mask, indices, fill):
+    if torch.version.hip is not None:
+        # Avoid a triton compiler failure
+        return fallback__unsafe_masked_index(self, mask, indices, fill)
```

Comment:
Do you have more logs or details about this failure?

Reply:
Fails with PassManager::run failed. Log at https://ossci-raw-job-status.s3.amazonaws.com/log/22537833056

Reply:
We can't really use this if it regresses ROCm support. From the logs you linked it looks like its the backward that's failing though, so perhaps we can still lower the forward?

BTW running CI with `torch._inductor.config.compile_threads = 1` might give a better compiler error. Some of the message gets lost when compiling in a process pool.

Reply:
It's a compiler failure. https://github.com/pytorch/pytorch/pull/121801 should fix this for us.